docs/development.rst

```diff
@@ -0,0 +1,283 @@
++--------------------------+--------------------+----------------------+-------+
+| OS-X                     |                    |                      |       |
++--------------------------+--------------------+----------------------+-------+
+| Windows                  |                    |                      |       |
++--------------------------+--------------------+----------------------+-------+
+
+
+.. _`Building the documentation`:
+
+Building the documentation
+--------------------------
+
+The ReadTheDocs (RTD) site is used to publish the documentation for the
+pywbem package at https://pywbem.readthedocs.io/
+
+This page is automatically updated whenever the Git repo for this package
+changes the branch from which this documentation is built.
+
+In order to build the documentation locally from the Git work directory,
+execute:
+
+::
+
+    $ make builddoc
+
+The top-level document to open with a web browser will be
+``build_doc/html/docs/index.html``.
+
+
+.. _`Testing`:
+
+.. # Keep the tests/README file in sync with this 'Testing' section.
+
+Testing
+-------
+
+
+All of the following `make` commands run the tests in the currently active
+Python environment. Depending on how the `pywbem` package is installed in that
+Python environment, either the `pywbem` and `pywbem_mock` directories in the
+main repository directory are used, or the installed `pywbem` package.
+The test case files and any utility functions they use are always used from
+the `tests` directory in the main repository directory.
+
+The `tests` directory has the following subdirectory structure:
+
+::
+
+    tests
+     +-- unit                Unit tests
+     |    +-- utils               Utility functions used by unit tests
+     +-- manual              Manual tests
+     +-- schema              The CIM schema MOF files used by some MOF tests
+
+There are multiple types of tests in pywbem:
+
+1. Unit tests and function tests
+
+   Today, the unit tests and function tests are contained in the single
+   directory `unit`. Unit tests do not require any WBEM server to be available,
+   and the tests validate their results automatically.
+
+   The distinction between unit tests and function tests as used in pywbemtools is
+   that function tests exercise the entire pywbemcli client component or entire
+   pywbem scripts using the pywbem mock module and mock CIM model definitions
+   to emulate a WBEM server, while unit tests exercise single modules without
+   using access to a WBEM server.
+
+
+   They are run by executing:
+
+   ::
+
+       $ make test
+
+   Test execution can be modified by a number of environment variables, as
+   documented in the make help (execute `make help`).
+
+3. Manual tests
+
+   There are several Python scripts and shell scripts that can be run manually.
+   The results need to be validated manually.
+
+   These scripts are in the directory:
+
+   ::
+
+       tests/manualtest/
```

Comment:
Inconsistent name of the manual test directory: `manual` vs `manualtest`. Decide for one and use consistently.

Reply:
DONE